src/Tmds.Ssh/KerberosCredential.cs

```diff
@@ -0,0 +1,339 @@
+    private static readonly byte[] KRB5_OID = [ 0x06, 0x09, 0x2A, 0x86, 0x48, 0x86, 0xF7, 0x12, 0x01, 0x02, 0x02 ];
+
+    private readonly NetworkCredential _kerberosCredential;
+    private readonly bool _delegateCredential;
+    private readonly string? _serviceName;
+
+#if NET8_0
+    // This API was made public in .NET 9 through ComputeIntegrityCheck.
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "GetMIC")]
+    private extern static void GetMICMethod(NegotiateAuthentication context, ReadOnlySpan<byte> data, IBufferWriter<byte> writer);
+#endif
+
+    /// <summary>
+    /// Create a credential for Kerberos authentication.
+    /// </summary>
+    /// <remarks>
+    /// The credential specified is used for the Kerberos authentication process. This can either be the same or
+    /// different from the username specified through <c>SshClientSettings.UserName</c>. The client settings username
+    /// is the target login user the SSH service is meant to run as, whereas the credential is the Kerberos
+    /// principal used for authentication. The rules for how a Kerberos principal maps to the target user is defined by
+    /// the SSH service itself. For example on Windows the username should be the same but on Linux the mapping can be
+    /// done through a <c>.k5login</c> file in the target user's home directory.
+    ///
+    /// If the credential is <c>null<c>, the Kerberos authentication will be done using a cached ticket.
+    /// For Windows, this is the current thread's identity (typically logon user) will be used.
+    /// For Unix/Linux, this will use the Kerberos credential cache principal, which may be managed using the
+    /// <c>kinit</c> command. If there is no available cache credential, the authentication will fail.
+    ///
+    /// Credentials can only be delegated if the Kerberos ticket retrieved from the KDC is marked as forwardable.
+    /// Windows hosts will always retrieve a forwardable ticket but non-Windows hosts may not. When using an explicit
+    /// credential, make sure that 'forwardable = true' is set in the krb5.conf file so that .NET will request a
+    /// forwardable ticket required for delegation. When using a cached ticket, make sure that when the ticket was
+    /// retrieved it was retrieved with the forwardable flag. If the ticket is not forwardable, the authentication will
+    /// still work but the ticket will not be delegated.
+    /// </remarks>
+    /// <param name="credential">The credentials to use for the Kerberos authentication exchange. Set to null to use a cached ticket.</param>
+    /// <param name="delegateCredential">Allows the SSH server to delegate the user on remote systems.</param>
+    /// <param name="serviceName">Override the service principal name (SPN), default uses the <c>host/<SshClientSettings.Host>.</c></param>
+    public KerberosCredential(NetworkCredential? credential = null, bool delegateCredential = false, string? serviceName = null)
```

Comment:
Can you consider the names and the order of these arguments, and feel free to change what you want if you think it improves the usability.

Reply:
We talked about the arg names earlier and I settled on these because they align with the Kerberos/GSSAPI nomenclature. I think the ordering makes sense here but is there something in particular you are concerned about.

Reply:
No concern from my side. I'm double checking because this is the public API and I find it important we have good usability.